### PR TITLE
feat: added button component

### LINKED ISSUE #2: Button component
**NOTE:** Use storybook to preview your component. Ideally do not create pages to test your component. If you do make on, remember to delete the page before pushing. Refer `src/components/text.tsx` and `src/stories/text.stories.tsx` for examples

## Variants
The button component will have the following variants:

### Primary
`primary`
Create red primary button, turns white on hover.
image

### Secondary
`secondary`
Create white secondary buttons, glow on hover
image

### Tertiary
`teritary`
Create greyish secondary button, slight brightness increase on hover
image

#### Primary
`ghost-primary`
image

#### Secondary
`ghost-secondary`
image

### PR DESCRIPTION
**Pull request by OneRandom1509**
fixes #2. Added all 5 variants.

This button component has 4 arguments:
1. variant = primary | secondary | tertiary | ghost-primary | ghost-secondary
2. text = text to be put in the button

3. className

### CODE REVIEW

File: src/components/button.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import Link from "next/link";
+
+const buttonVariants = cva(
+  "px-6 py-3 font-orbitron text-lg font-bold rounded-[16px] transition-all",
+  {
+    variants: {
+      variant: {
+        primary:
+          "border border-red-500 border-2 text-red-500 bg-transparent shadow-[_inset_0_0_15px_#FF000080,_0_0_15px_#FF000040] hover:text-white [text-shadow:_0_0_10px_#FF0000,0_0_20px_#FF0000,0_0_30px_#FF0000]",
+        secondary:
```
Comment (ThEditor) [inline src/components/button.tsx:12]: Secondary on hover should turn to red (same color as primary)
Affect the borders as well

File: src/components/button.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          "border border-gray-400 border-2 text-gray-400 bg-transparent hover:text-white hover:shadow-[inset_0_0_15px_#ffffff80,0_0_15px_#ffffff80,0_0_40px_#ffffff40,0_0_60px_#ffffff20] hover:brightness-125 [text-shadow:_0_0_10px_#ffffff80,0_0_15px_#ffffff40]",
+        tertiary:
+          "border border-gray-500 border-2 text-gray-400 hover:text-gray-200 hover:brightness-125",
+        "ghost-primary":
+          "text-red-500 hover:text-white [text-shadow:_0_0_10px_#FF0000,0_0_20px_#FF0000,0_0_30px_#FF0000]",
+        "ghost-secondary":
```
Comment (ThEditor) [inline src/components/button.tsx:18]: Secondary on hover should turn to red (same color as primary)
Affect the borders as well

File: src/components/button.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  ({ text, className, variant = "primary", href, ...other }, ref) => {
+    const commonProps = {
+      ref,
+      className: cn(buttonVariants({ variant }), className),
+      ...other,
+    };
+
+    if (href) {
+      return (
+        <Link href={href} passHref>
```
Comment (ThEditor) [inline src/components/button.tsx]: Expected is a button, not a link
Created a separate compenent "LinkButton" for this (copy the same styles as `ghost-primary` and `ghost-secondary` but call them `primary` and `secondary`)

**Review by ThEditor — CHANGES_REQUESTED** [review]
Everything looks (almost) perfect to me, nicely done!

File: src/components/button.tsx (hunk 1 above)
Comment (OneRandom1509) [reply src/components/button.tsx:12]: needed changes have been done

File: src/components/button.tsx (hunk 3 above)
Comment (OneRandom1509) [reply src/components/button.tsx]: removed link :+1:

**Review by ThEditor — APPROVED** [review]
lgtm

**Pull request merged**